project/project.go

```diff
@@ -238,21 +265,36 @@ func (p *Project) load(file string, bytes []byte) error {
 			}
 		}
 	}
+}
 
-	// FIXME(vdemeester) Not sure about this..
-	if p.context.NetworksFactory != nil {
-		networks, err := p.context.NetworksFactory.Create(p.Name, p.NetworkConfigs, p.ServiceConfigs, p.isNetworkEnabled())
-		if err != nil {
-			return err
-		}
+func (p *Project) isNetworkEnabled() bool {
+	return p.configVersion == "2"
+}
 
-		p.networks = networks
+func (p *Project) handleVolumeConfig() {
+	if p.isVolumeEnabled() {
+		for _, serviceName := range p.ServiceConfigs.Keys() {
+			serviceConfig, _ := p.ServiceConfigs.Get(serviceName)
+			// Consolidate the name of the volume
+			// FIXME(vdemeester) probably shouldn't be there, maybe move that to interface/factory
```

Comment:
yes, same comment here and for network, this logic will have to be extracted at some point to be easier to customize :stuck_out_tongue_closed_eyes: